File: src/org/ice1000/uset/UnionSet.java
package org.ice1000.uset;

/**
 * Created by ice1000 on 2016/11/17.
 *
 * @author ice1000
 */
abstract class UnionSet {
	protected native int find(long[] data, int length, int num);

	protected native void merge(long[] data, int length, int a, int b);
}

File: src/main/kotlin/org/algo4j/UnionSet.kt
@file:JvmMultifileClass
@file:JvmName("Extension")

/**
 * Created by ice1000 on 2017/1/4.
 *
 * @author ice1000
 */
package org.algo4j

import org.algo4j.tree.UnionSet
import org.jetbrains.annotations.Contract

@Contract(pure = false)
operator fun UnionSet.get(index: Int) =
		find(index)
